assimilation_code/modules/utilities/parse_args_mod.f90

```diff
@@ -554,7 +587,407 @@ subroutine get_next_arg(inline, startoff, argword, endoff)
+
+type(csv_file_type), intent(in)  :: cf
+character(len=*),    intent(in)  :: varname
+integer,             intent(out) :: varvals(:)
+character(len=*),    intent(in), optional :: context
+
+character(len=*), parameter :: routine = 'csv_get_field_int'
+
+integer            :: i
+character(len=512) :: strvals(size(varvals))
+
+! Read it as string first
+call csv_get_field_char(cf, varname, strvals, context)
+
+! Then convert to int
+do i = 1, size(varvals)
+   varvals(i) = string_to_integer(strvals(i))
+enddo
+
+end subroutine csv_get_field_int
+
+
+!--------------------------------------------------------------
+! Read real data from csv file
+subroutine csv_get_field_real(cf, varname, varvals, context)
+
+type(csv_file_type), intent(in)  :: cf
+character(len=*),    intent(in)  :: varname
+real(r8),            intent(out) :: varvals(:)
+character(len=*),    intent(in), optional :: context
+
+character(len=*), parameter :: routine = 'csv_get_field_real'
+
+integer            :: i
+character(len=512) :: strvals(size(varvals))
+
+call csv_get_field_char(cf, varname, strvals, context)
+
+! To real for all column data
+do i = 1, size(varvals)
+   varvals(i) = string_to_real(strvals(i))
+enddo
+
+end subroutine csv_get_field_real
+
+
+!----------------------------------------------------
+! Get number of rows in the csv file (without header line) 
+integer function csv_get_obs_num(fname, context) result(nrows)
+
+character(len=*), intent(in) :: fname
+character(len=*), intent(in), optional :: context
+
+character(len=*), parameter  :: routine = 'csv_get_obs_num'
+
+! Number of data entries in file
+call find_textfile_dims(fname, nrows)
+
+! Omit the header
+nrows = nrows - 1
+
+! Check if no data in file
+if (nrows <= 0) then 
+   write(string1, '(4A, i0)') 'Input CSV file "', trim(fname), '" contains ', &
+           'no data. The number of data lines is ', nrows
+   call error_handler(E_ERR, routine, string1, context)
+endif
+
+end function csv_get_obs_num
+
+
+!---------------------------------------------------
+! Adapt get_args_from_string after adjusting delims
+subroutine split_fields(line, delim, nfields, fields)
+
+character(len=*), intent(in)  :: line
+character,        intent(in)  :: delim
+integer,          intent(out) :: nfields
+character(len=*), intent(out) :: fields(:)
+
+character(len=MAX_FIELDS_LEN) :: work
+
+! Clean the line then parse it 
+work = normalize_delims(line, delim)
+call get_args_from_string(work, nfields, fields)
+
+end subroutine split_fields
+
+
+!----------------------------------------------------------------------
+! Replace ',' and ';' with blanks to use above parsers. 
+! We also need to treat empty fields so that we don't
+! collapse with the spaces and cause any column drifts. 
+! This serves as a wrapper for 'get_args_from_string'
+! Example: 
+! A;B;;;;C;; --> A B _EMPTY_ _EMPTY_ _EMPTY_ C _EMPTY_ _EMPTY_
+function normalize_delims(line, delim) result(out_line)
```

Comment:
i think the longer term solution here is to update get_args_from_string() so you can send in a specific delimiter character instead of assuming blanks.  but for now, this will solve the need.